src/adapter/src/coord/timestamp_oracle/postgres_oracle.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A timestamp oracle backed by "Postgres" for persistence/durability and where
+//! all oracle operations are self-sufficiently linearized, without requiring
+//! any external precautions/machinery.
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use deadpool_postgres::{Object, PoolError};
+use mz_ore::metrics::MetricsRegistry;
+use mz_postgres_client::metrics::PostgresClientMetrics;
+use mz_postgres_client::{PostgresClient, PostgresClientConfig, PostgresClientKnobs};
+use mz_repr::Timestamp;
+use tracing::{debug, info};
+
+use crate::coord::timeline::WriteTimestamp;
+use crate::coord::timestamp_oracle::{GenericNowFn, TimestampOracle};
+
+// These `sql_stats_automatic_collection_enabled` are for the cost-based
+// optimizer but all the queries against this table are single-table and very
+// carefully tuned to hit the primary index, so the cost-based optimizer doesn't
+// really get us anything. OTOH, the background jobs that crdb creates to
+// collect these stats fill up the jobs table (slowing down all sorts of
+// things).
+//
+// WIP: Do we need this for the postgres oracle? This is originally from
+// persist.
+const SCHEMA: &str = "
+CREATE TABLE IF NOT EXISTS timestamp_oracle (
+    timeline text NOT NULL,
+    read_ts bigint NOT NULL,
+    write_ts bigint NOT NULL,
+    PRIMARY KEY(timeline)
+) WITH (sql_stats_automatic_collection_enabled = false);
```

Comment:
Yeah, same, but I chose to use "postgres" because that's what all the other places use. I think I left a snarky comment in a commit message somewhere ... :sweat_smile: